airflow/serialization/schema.json

```diff
@@ -102,7 +102,8 @@
         "_task_group": {"anyOf": [
           { "type": "null" },
           { "$ref": "#/definitions/task_group" }
-        ]}
+        ]},
+        "edge_info": { "$ref": "#/definitions/dict" }
```

Comment:
Oh this could maybe be specified a bit tighter. WDYT @kaxil ?

Reply:
Yea similar to how we only store `_downstream_task_ids` (not the `_upstream_task_ids`) for a task and we can set upstream for the other when de-serializing. We do that to keep the size of serialized blog as small as possible.



https://github.com/apache/airflow/blob/8cc8d11fb87d0ad5b3b80907874f695a77533bfa/airflow/serialization/serialized_objects.py#L451-L452

https://github.com/apache/airflow/blob/8cc8d11fb87d0ad5b3b80907874f695a77533bfa/airflow/serialization/serialized_objects.py#L666-L668

https://github.com/apache/airflow/blob/8cc8d11fb87d0ad5b3b80907874f695a77533bfa/airflow/serialization/serialized_objects.py#L724-L727

https://github.com/apache/airflow/blob/8cc8d11fb87d0ad5b3b80907874f695a77533bfa/airflow/models/baseoperator.py#L1365-L1370

We pay the price of hardcoding it but it is worth it as we can save MBs (which would be transmitted to and from the database) when number of tasks is huge.

Reply:
Sure, I'll add a slightly more compact representation. I can't save any actual data, since source-target-label is all that's stored, but we can save the outside wrapping with `_type` etc. at least?